### ISSUE TITLE
(BSP) Importing existing project as BSP model fails on build

### ISSUE THREAD

**Issue by ianknowles** [sbt/sbt#6450]
After importing an existing project in IntelliJ IDEA 2021.1 (one that I have previously used for BSP testing), the wizard freezes with the message `building ...` and `sbt:PROJECTNAME>`. It never completes this step, (likely the build has failed), but no error message is given, the import does not back out, and pressing cancel does nothing, IDEA has to be force closed as well as a background jvm process thats presumably still hanging the build. At minimum there should be an error message here and the import terminated.

The project is set to sbt version 1.4.7 in `build.properties` which I was previously testing. The `sbt.json` created points to the IntelliJ managed version of sbt which wasn't fully supported until a bug fix in 1.5.0. However setting the project to 1.5.0 still results in the same import hang. The sbt Launcher option for new projects is already set to a standalone sbt install used in my previous testing so the import is ignoring this option and opting for the bundled sbt, possibly another issue. Editing the `sbt.json` to use a standalone install has no effect and the project is unable to run the build (previously 2020.2 and sbt 1.4.7 were able to build the project).

**Comment by adpi2** [comment]
Thanks for reporting!

It's really hard to tell what's going on here because I don't recognize the logs: `building ...` and `sbt:PROJECTNAME>`. It is like if the `sbt.json` command has started an sbt shell rather than a BSP client.

I tried to reproduce by downgrading sbt from 1.5.0 to 1.4.9 and indeed it does not work but the IntelliJ messaging is different:
![image](https://user-images.githubusercontent.com/13123162/114514929-1c32e380-9c3c-11eb-8106-570b52e698e6.png)

The reason is that the new `--sbt-launch-jar` option derails sbt 1.4.9:
```shell
> "/home/piquerez/.cache/coursier/jvm/adopt@1.8.0-275/jre/bin/java" "-Xms100m" "-Xmx100m" "-classpath" "/home/piquerez/.local/share/JetBrains/Toolbox/apps/IDEA-C/ch-0/203.7717.56.plugins/Scala/launcher/sbt-launch.jar" "xsbt.boot.Boot" "-bsp" "--sbt-launch-jar=/home/piquerez/.local/share/JetBrains/Toolbox/apps/IDEA-C/ch-0/203.7717.56.plugins/Scala/launcher/sbt-launch.jar"
[info] server was not detected. starting an instance
[info] welcome to sbt 1.4.9 (AdoptOpenJDK Java 1.8.0_275)
[info] loading global plugins from /home/piquerez/.sbt/1.0/plugins
[info] loading settings for project scaladex-build from plugins.sbt ...
[info] loading project definition from /home/piquerez/scalacenter/scaladex/scaladex/project
[warn] There may be incompatibilities among your library dependencies; run 'evicted' to see detailed eviction warnings.
[info] compiling 2 Scala sources to /home/piquerez/scalacenter/scaladex/scaladex/project/target/scala-2.12/sbt-1.0/classes ...
[info] done compiling
[info] loading settings for project scaladex from build.sbt ...
[info] resolving key references (10498 settings) ...
[info] set current project to scaladex (in build file:/home/piquerez/scalacenter/scaladex/scaladex/)
[error] Expected symbol
[error] Not a valid command: --
[error] Expected 'debug'
[error] Expected 'info'
[error] Expected 'warn'
[error] Expected 'error'
[error] Expected 'addPluginSbtFile'
[error] --sbt-launch-jar=/home/piquerez/.local/share/JetBrains/Toolbox/apps/IDEA-C/ch-0/203.7717.56.plugins/Scala/launcher/sbt-launch.jar
[error]   ^
``` 

I don't think we should do anything about it.

**Comment by adpi2** [comment]
@ianknowles can you reproduce the bug? If so, can you run the `sbt.json` command manually and share the logs with us?

**Comment by ianknowles** [comment]
Sorry my description was a little off, its not after importing but during the import process. As a final step of the import its running the sbt build (I assume to extract the project structure and generate the IDEA modules and config files). Before starting the import I delete the .bsp folder and the .idea.

![image](https://user-images.githubusercontent.com/26507830/114555174-595c9d00-9c5f-11eb-92ec-ffba50cf8419.png)

![image](https://user-images.githubusercontent.com/26507830/114555310-78f3c580-9c5f-11eb-9dd2-742b415419bd.png)

After further testing there was a error in the sbt build, I had renamed the subprojects in the build and the final step is meant to activate the main project, but still contained a reference to an old name. So there was a user configuration error in the build preventing it from finishing, but as I mentioned in the original report at this point while importing a project the build should not hang and block the IDE, it should print an error message and terminate.

![image](https://user-images.githubusercontent.com/26507830/114567354-11dc0e00-9c6b-11eb-919f-3b83199250f1.png)

**Comment by ianknowles** [comment]
Heres an example `build.sbt` that will hang the import. The problem is the final line `"project servers"`, if this is corrected to `"project server"` then the build finishes and the project imports successfully. Otherwise it hangs on the Setting up BSP configuration and doesn't respond to cancel etc.

```scala
name := """test"""

version := "1.0-SNAPSHOT"

lazy val server = (project in file("server"))
	.settings(commonSettings)
	.settings(
		scalaJSProjects := Seq(client),
		pipelineStages in Assets := Seq(scalaJSPipeline),
		pipelineStages := Seq(digest, gzip),
		// triggers scalaJSPipeline when using compile or continuous compilation
		compile in Compile := ((compile in Compile) dependsOn scalaJSPipeline).value,
		libraryDependencies ++= Seq(
			"com.vmunier" %% "scalajs-scripts" % "1.1.4",
			guice,
			specs2 % Test,
			"org.scalatestplus.play" %% "scalatestplus-play" % "5.1.0" % Test,
			"org.webjars" %% "webjars-play" % "2.8.0",
		),
	)
	.enablePlugins(PlayScala)
	.dependsOn(sharedJvm)

lazy val client = (project in file("client"))
	.settings(commonSettings)
	.settings(
		scalaJSUseMainModuleInitializer := true,
		mainClass in compile := Some("graphs.Main"),
		libraryDependencies ++= Seq(
			"org.scala-js" %%% "scalajs-dom" % "1.1.0",
		)
	)
	.enablePlugins(ScalaJSPlugin, ScalaJSWeb)
	.dependsOn(sharedJs)

lazy val shared = crossProject(JSPlatform, JVMPlatform)
	.crossType(CrossType.Pure)
	.in(file("shared"))
	.settings(commonSettings)

lazy val sharedJvm = shared.jvm
lazy val sharedJs = shared.js

lazy val commonSettings = Seq(
	scalaVersion := "2.13.1",
	scalacOptions ++= Seq(
		"-feature",
		"-deprecation",
		"-Xfatal-warnings",
		"-target:11"
	),
	javacOptions ++= Seq("-source", "11", "-target", "11")
)

// loads the server project at sbt startup
onLoad in Global := (onLoad in Global).value.andThen(state => "project servers" :: state)
```

**Comment by adpi2** [comment]
Thanks for the repro!

I am closing the issue because it should be fixed on IntelliJ side: https://youtrack.jetbrains.com/issue/SCL-18940
